src/main/java/essenmakanan/parser/ShortcutParser.java

```diff
@@ -69,25 +73,37 @@ private static void editShortcutName(Shortcut shortcut, IngredientList ingredien
     }
 
     private static void editShortcutQuantity(Shortcut shortcut, String editDetail, boolean hasEditQuantity)
-            throws EssenEditShortcutException {
+            throws EssenEditShortcutException, NumberFormatException {
         double newQuantity = Double.parseDouble(editDetail.substring(2).strip());
 
         if (hasEditQuantity) {
             throw new EssenEditShortcutException("usage");
         }
 
+        if (!checkForValidQuantity(newQuantity)) {
+            throw new NumberFormatException();
```

Comment:
should we be keeping the exceptions to our Essen exceptions only?

Reply:
this will trigger essen exception after throwing it